### PR TITLE
gh-93951: In test_bdb.StateTestCase.test_skip, avoid including auxiliary importers.

### LINKED ISSUE #93951: test_bdb is sensitive to sys.meta_path manipulation
Here's a repro not involving Setuptools showing how `test_bdb` is sensitive to `sys.meta_path` manipulation:

```
FROM jaraco/multipy-tox

RUN apt install -y libarchive-tools
RUN apt install -y build-essential gdb lcov pkg-config libbz2-dev libffi-dev libgdbm-dev libgdbm-compat-dev liblzma-dev libncurses5-dev libreadline6-dev libsqlite3-dev libssl-dev lzma lzma-dev tk-dev uuid-dev zlib1g-dev
RUN wget -O - https://github.com/python/cpython/archive/refs/heads/3.9.zip | bsdtar -xf -
WORKDIR cpython-3.9
RUN sh ./configure --prefix /opt/python
RUN make install
WORKDIR /
RUN /opt/python/bin/python3 -m pip install --upgrade pip
RUN /opt/python/bin/python3 -m pip uninstall -y setuptools
RUN echo "import sys; sys.meta_path.insert(0, type('NullFinder', (), dict(find_spec=lambda *args: None))())" > /opt/python/lib/python3.9/site-packages/null-finder.pth
CMD /opt/python/bin/python3 -m test_bdb
```

It fails with a similar error:

```
AssertionError: Wrong event type at expect_set item 2, got 'call'
  Expected: ('line', 3, 'tfunc_import')
  Got:      ('call', 1, '<lambda>'),        ('quit',),
```

_Originally posted by @jaraco in https://github.com/python/cpython/issues/91169#issuecomment-1158978878_

### PR DESCRIPTION
**Pull request by jaraco**
Fixes gh-93951

### CODE REVIEW

File: Lib/test/support/__init__.py [hunk 1]
```diff
@@ -1768,6 +1768,18 @@ def cleanup():
     setattr(object_to_patch, attr_name, new_value)
 
 
+@contextlib.contextmanager
+def patch_list(orig):
+    """
+    Like unittest.mock.patch.dict, but for lists.
+    """
```
Comment (brettcannon) [inline Lib/test/support/__init__.py]:
```suggestion
    """Like unittest.mock.patch.dict, but for lists."""
```

**Review by brettcannon — APPROVED** [review]
One PEP 8 violation, but otherwise LGTM!

**Comment by tiran** [comment]
Why is the new helper method needed? The case looks like a job for https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch

**Comment by ambv** [comment]
@serhiy-storchaka points out that the new `patch_list` helper preserves identity of the patched list which might be an important property for some use cases. In particular, looks like `sys.meta_path` manipulation in tests always preserves the list's identity. Let's keep this then.

**Pull request merged**